src/Raven.Client/Json/BlittableJsonReader.cs

```diff
@@ -151,7 +151,20 @@ private bool SetToken(BlittableJsonToken token, object value)
                     SetToken(JsonToken.Integer, (long)value);
                     return true;
                 case BlittableJsonToken.LazyNumber:
-                    SetToken(JsonToken.Float, (double)((LazyNumberValue)value));
+                    var lazyNumber = (LazyNumberValue)value;
+                    var asDouble = (double)lazyNumber;
+                    var asDoubleString = asDouble.ToString("G17",CultureInfo.InvariantCulture);
```

Comment:
Not acceptable.
This requires an _expensive_ string allocation & formatting here.